datastream-server/src/main/java/com/linkedin/datastream/server/zk/ZkAdapter.java

```diff
@@ -798,24 +797,31 @@ private void issueAssignmentTokensForStoppingDatastreams(List<DatastreamGroup> s
         for (String instance : instances) {
           String assignmentTokenPath = KeyBuilder.datastreamAssignmentTokenForInstance(_cluster,
               stoppingStream.getName(), instance);
-          AssignmentToken token = new AssignmentToken(hostname, instance);
+          AssignmentToken token = new AssignmentToken(_instanceName, instance, System.currentTimeMillis());
           _zkclient.create(assignmentTokenPath, token.toJson(), CreateMode.PERSISTENT);
         }
       }
     }
   }
 
   /**
-   * Gets the name of the local host
+   * Claims assignment tokens for the given datastreams and instance
+   * @param datastreams List of datastreams for which tokens are to be claimed
+   * @param instance Instance name
    */
-  private String getLocalHostName() {
-    String hostname = "localhost";
-    try {
-      hostname = InetAddress.getLocalHost().getHostName();
-    } catch (UnknownHostException ex) {
-      LOG.warn("Unable to obtain hostname for leader");
+  public void claimAssignmentTokensForDatastreams(List<Datastream> datastreams, String instance) {
+    for (Datastream stream : datastreams) {
+      String streamName = stream.getName();
+      String tokenPath = KeyBuilder.datastreamAssignmentTokenForInstance(_cluster, streamName, instance);
+      if (_zkclient.exists(tokenPath)) {
```

Comment:
You need the `else` path. Also add more logging around this.

Reply:
I didn't have an "else" path for this because I wanted it to be a no-op (i.e. attempt to claim non-existing token). Will add a debug log in the else path.